plugin-examples/src/main/java/com/hotels/styx/ExampleMessageIntercept.java

```diff
@@ -0,0 +1,78 @@
+package com.hotels.styx;
+
+import com.hotels.styx.api.Eventual;
+import com.hotels.styx.api.HttpResponse;
+import com.hotels.styx.api.LiveHttpRequest;
+import com.hotels.styx.api.LiveHttpResponse;
+import com.hotels.styx.api.plugins.spi.Plugin;
+
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_TYPE;
+import static com.hotels.styx.api.HttpResponseStatus.OK;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * This shows an example of an inteceptor that responds to a received request
+ */
+
+public class Example implements Plugin {
+
+    @Override
+    public Eventual<LiveHttpResponse> intercept(LiveHttpRequest request, Chain chain) {
+        if (request.header("X-Respond").isPresent()) {
+            // request.consume();
```

Comment:
There should not be any commented code here.

Also please add a comment or two explaining what this code is doing - e.g. explain the branching condition and what happens in each branch.